frontend/src/components/pages/WaitlistExperience/index.tsx

```diff
@@ -1,35 +1,117 @@
+waitListNextStepTxtMap.set(WaitListExperienceSteps.PersonalInfoPage, "Submit");
+
+const WaitlistExperiencePage = ({
+  camp,
+  selectedCampSessions,
+  onClickBack,
+}: WaitlistExperiencePageProps): React.ReactElement => {
+  const nextBtnRef = useRef<HTMLButtonElement>(null);
+  const [currentStep, setCurrentStep] = useState<WaitListExperienceSteps>(
+    WaitListExperienceSteps.PersonalInfoPage,
+  );
+  const [campers, setCampers] = useState<CreateWaitlistedCamperDTO[]>([
+    {
+      firstName: "",
+      lastName: "",
+      age: NaN,
+      contactEmail: "",
+      contactFirstName: "",
+      contactLastName: "",
+      contactNumber: "",
+    },
+  ]);
 
-const WaitlistExperiencePage = (): React.ReactElement => {
-  const [isConfirmationStep, setIsConfirmationStep] = useState(false);
+  if (!camp)
+    return (
+      <Center>
+        <Spinner />
+      </Center>
+    );
 
-  // Temp variables for personal info page
-  const [isPersonalInfoChecked, setIsPersonalInfoChecked] = useState(false);
+  const handleStepNavigation = (stepsToMove: number) => {
+    const desiredStep = currentStep + stepsToMove;
+    if (WaitListExperienceSteps[desiredStep]) {
+      setCurrentStep(currentStep + stepsToMove);
+    } else if (desiredStep < 0) {
+      onClickBack();
+    }
+  };
 
-  const isPersonalInfoComplete = isPersonalInfoChecked;
+  const isCurrentStepCompleted = (step: WaitListExperienceSteps) => {
+    switch (step) {
+      case WaitListExperienceSteps.PersonalInfoPage:
+        return checkPersonalInfoFilled(campers, camp);
+      default:
+        return false;
+    }
+  };
 
-  const submitHandler = () => {
-    if (!isPersonalInfoComplete) {
-      alert("PLACEHOLDER - personal info step not completed");
-    } else {
-      setIsConfirmationStep(true);
+  const getCurrentRegistrantStepComponent = (step: WaitListExperienceSteps) => {
+    switch (step) {
+      case WaitListExperienceSteps.PersonalInfoPage:
+        return (
+          <WaitlistPersonalInfoPage
+            nextBtnRef={nextBtnRef}
+            campers={campers}
+            camp={camp}
+            setCampers={setCampers}
+          />
+        );
+      case WaitListExperienceSteps.ConfirmationPage:
+        return (
+          <WaitlistConfirmationPage
+            campSessions={selectedCampSessions}
+            campers={campers}
+            campName={camp.name}
+          />
+        );
+      default:
+        throw new Error("unexpected page");
     }
   };
+
   return (
-    <>
-      {isConfirmationStep ? (
-        <WaitlistConfirmationPage />
-      ) : (
-        <WaitlistPersonalInfoPage
-          isChecked={isPersonalInfoChecked}
-          toggleChecked={() => setIsPersonalInfoChecked(!isPersonalInfoChecked)}
-        />
-      )}
-      {!isConfirmationStep && <WaitlistFooter submitHandler={submitHandler} />}
-    </>
+    <Box>
+      <EmptyNavBar />
+      <Flex
+        direction="column"
+        w="100vw"
```

Comment:
Delete the w="100vw", this is what causes the strange horizontal scroll to occur.